wait/host.go

```diff
@@ -158,18 +158,10 @@ func (a *HostAwaitility) WaitForDeletedMasterUserRecord(name string) error {
 	})
 }
 
-func getUaSpecSyncIndex(mur *toolchainv1alpha1.MasterUserRecord, targetCluster string) string {
-	for _, ua := range mur.Spec.UserAccounts {
-		if ua.TargetCluster == targetCluster {
-			return ua.SyncIndex
-		}
-	}
-	return ""
-}
-
-func containsUserAccountStatus(uaStatuses []toolchainv1alpha1.UserAccountStatusEmbedded, uaStatus toolchainv1alpha1.UserAccountStatusEmbedded) bool {
-	for _, status := range uaStatuses {
-		if reflect.DeepEqual(uaStatus, status) {
+func containsUserAccountStatus(expectedStatuses []toolchainv1alpha1.UserAccountStatusEmbedded, actualStatus toolchainv1alpha1.UserAccountStatusEmbedded) bool {
+	for _, expectedStatus := range expectedStatuses {
+		if expectedStatus.TargetCluster == actualStatus.TargetCluster &&
+			test.ConditionsMatch(expectedStatus.Conditions, actualStatus.Conditions...) {
```

Comment:
don't forget that we need to verify also the syncIndex - that was the reason for having the `DeepEqual` there - to verify all params.

Reply:
I had issues with the `DeepEqual` func when comparing the status, because of the `LastTransitionTime` field which we can't know in advance. Besides, I don't understand what we reallt try to verify here with the `synIndex` field, since we first get it from the `actual` MUR, then set it in the `expected` UserAccountStatus to finally compare both.

Reply:
> I had issues with the DeepEqual func when comparing the status, because of the LastTransitionTime field which we can't know in advance. 

Yeah, I can imagine that when we have all the pieces together, then the status of the `UserAccount` is modified more often and the current code failed. But the fix shouldn't be removing the verification, but changing the flow the resources are verified in as well as the expected value.

> Besides, I don't understand what we reallt try to verify here with the synIndex field

we need to verify that the status synchronization between MUR and UserAccount works as expected

> since we first get it from the actual MUR, then set it in the expected UserAccountStatus to finally compare both. 

I'm not sure which part of the code you are referring to. What I can see is that it at first [get](https://github.com/codeready-toolchain/toolchain-e2e/blob/master/test/e2e/e2e_test.go#L198) the current UserAccount, then using the current status it [creates](https://github.com/codeready-toolchain/toolchain-e2e/blob/master/test/e2e/e2e_test.go#L199-L201) the expected `UserAccountStatusEmbedded` and then it uses it for [verification](https://github.com/codeready-toolchain/toolchain-e2e/blob/master/test/e2e/e2e_test.go#L206)

Reply:
fixed in 2d8cf14